src/components/modules/banners/BannerTypes.ts

```diff
@@ -40,19 +44,24 @@ export interface BannerVariant {
     moduleName: string;
 }
 
+export type BannerType = 'contributions' | 'subscriptions';
 export type CanRun = (targeting: BannerTargeting, pageTracking: BannerPageTracking) => boolean;
+export type BannerAudience = 'NonSupporters' | 'Supporters' | 'All';
 
 export interface BannerTest {
     name: string;
+    path: string;
+    bannerType: BannerType;
+    testAudience: BannerAudience;
     canRun: CanRun;
     minPageViews: number;
-    variants: BannerVariant[];
+    variants?: BannerVariant[];
 }
 
 // The result of selecting a test+variant for a user
 export interface BannerTestSelection {
     test: BannerTest;
-    variant: BannerVariant;
+    variant: BannerVariant | null;
```

Comment:
Thinking about this more, I really think there should always be a variant.
It would make this PR simpler, as there are several places where you're now handling the possible absence of a variant. E.g. we use the variant name to construct the campaign code.
I also don't know how the tracking is going to work if there's no variant!
Every banner is a `BannerTest`, even if we're not A/B testing it. And if it's not an A/B test then we just have a single 'control' variant .

Reply:
We talked this through too, and we agree. We'll change it back so there is always a variant.

Reply:
thanks!